### PR TITLE
Rashad rooms

### PR DESCRIPTION
**Pull request by rashadg1030**
Resolved the issues. New Door type. Added more fields to PlayState to accomadate walls and doors. Refactored wallTileMap and doorTileMap accordingly.

### CODE REVIEW

File: src/Day1.hs [hunk 1]
```diff
@@ -91,11 +93,14 @@ updatePlayerPos Input{keys=keys} (x,y) = (x',y')
 tileMap :: State -> Map (Int, Int) Tile
 tileMap st = case scene st of
   Scene'Start -> text "DAY 1" White1 (screenW `div` 2 - 2, screenH `div` 2)
-  Scene'Play -> mergeTiles clear $ fromList [ ( (x,y), Tile Nothing Nothing (Just wh1) ) ]
+  Scene'Play -> mergeTiles playerTile ( mergeTiles testDoors ( mergeTiles testWalls clear ) )
```
Comment (jxv) [inline src/Day1.hs]: The order is backwards. It should be `mergeTiles clear (` ....

File: src/Day1.hs [hunk 2]
```diff
@@ -113,6 +118,15 @@ clearTileMap c = fromList
 colorFromRoomIndex :: RoomIndex -> Color
 colorFromRoomIndex (RoomIndex idx) = colors !! (idx `mod` len)
   where
-    colors = colorWheel2
+    colors = rainbow
```
Comment (jxv) [inline src/Day1.hs]: use `colorWheel2`

File: src/Day1.hs [hunk 3]
```diff
@@ -113,6 +118,15 @@ clearTileMap c = fromList
...
+    
+  -- Tried to use TupleSections pragma but it wouldn't work
+wallTileMap :: [(Int, Int)] -> Map (Int, Int) Tile
+wallTileMap locs = fromList $ map (swap . (,) (Tile Nothing Nothing (Just bk2))) locs 
 
+doorTileMap :: [Int] -> [(Int, Int)] -> Map (Int, Int) Tile
+doorTileMap ids locs = fromList $ zipWith makeDoorTile ids locs
+
+makeDoorTile :: Int -> (Int, Int) -> ((Int, Int), Tile)
+makeDoorTile id loc = (,) loc $ Tile ( Just (intToDigit id, bk2) ) ( Just (Square, bk2) ) Nothing
```
Comment (jxv) [inline src/Day1.hs]: needs newline

File: src/Day1.hs [hunk 4]
```diff
@@ -113,6 +118,15 @@ clearTileMap c = fromList
...
     len = length colors
+    
+  -- Tried to use TupleSections pragma but it wouldn't work
+wallTileMap :: [(Int, Int)] -> Map (Int, Int) Tile
+wallTileMap locs = fromList $ map (swap . (,) (Tile Nothing Nothing (Just bk2))) locs 
 
+doorTileMap :: [Int] -> [(Int, Int)] -> Map (Int, Int) Tile
+doorTileMap ids locs = fromList $ zipWith makeDoorTile ids locs
+
+makeDoorTile :: Int -> (Int, Int) -> ((Int, Int), Tile)
```
Comment (jxv) [inline src/Day1.hs]: move this to a where clause under `doorTileMap`

File: src/Day1.hs [hunk 5]
```diff
@@ -91,11 +93,14 @@ updatePlayerPos Input{keys=keys} (x,y) = (x',y')
...
 tileMap st = case scene st of
   Scene'Start -> text "DAY 1" White1 (screenW `div` 2 - 2, screenH `div` 2)
-  Scene'Play -> mergeTiles clear $ fromList [ ( (x,y), Tile Nothing Nothing (Just wh1) ) ]
+  Scene'Play -> mergeTiles playerTile ( mergeTiles testDoors ( mergeTiles testWalls clear ) )
   Scene'GameOver -> fromList []
   where
     (x,y) = playerPos (player (playState st))
     clear = clearTileMap (colorFromRoomIndex (playerRoom (player (playState st))))
+    playerTile = fromList [ ( (x,y), Tile Nothing Nothing (Just wh1) ) ]
+    testDoors = doorTileMap [1, 2, 3] [(9,8), (10, 13), (3, 7)]
```
Comment (jxv) [inline src/Day1.hs]: The data for doors and walls should be coming from somewhere in the `State`

**Review by jxv — CHANGES_REQUESTED** [review]

File: src/Day1.hs (hunk 1 above)
Comment (rashadg1030) [reply src/Day1.hs]: When it was the reverse the player would go under the room tiles. I'll test it again. Maybe I was doing something wrong.

File: src/Day1.hs (hunk 2 above)
Comment (rashadg1030) [reply src/Day1.hs]: Okay will do. Going to fix now real quick

File: src/Day1.hs (hunk 4 above)
Comment (rashadg1030) [reply src/Day1.hs]: Okay good idea.

File: src/Day1.hs (hunk 5 above)
Comment (rashadg1030) [reply src/Day1.hs]: Okay will do. I will create a new Door datatype perhaps.

Comment (jxv) [reply src/Day1.hs]: If there are just list of indices they don't need a special type.

Comment (jxv) [reply src/Day1.hs]: Wait. Nevermind.

**Comment by jxv** [comment]
Except for the door number, everything is good.

**Pull request merged**